src/code/templates/fv-egg-sort-game.js

```diff
@@ -330,16 +342,17 @@ export default class FVEggSortGame extends Component {
   }
 
   handleBasketClick = (basketIndex, basket) => {
-    const { correct, errors, onSubmitEggForBasket } = this.props,
-          { eggs } = this.state,
-          { index: selectedEggIndex, egg: selectedEgg } = this.selectedEgg(),
-          eggDrakeIndex = selectedEggIndex,
-          isChallengeComplete = correct + errors + 1 >= eggs.length;
+    const { onSubmitEggForBasket } = this.props,
+          { index: selectedEggIndex, egg: selectedEgg } = this.selectedEgg();
     isSubmittedEggCorrect = selectedEgg && isEggCompatibleWithBasket(selectedEgg, basket);
     if (selectedEgg) {
       animationEvents.moveEggToBasket.animate(selectedEgg, selectedEggIndex, basketIndex);
-      setTimeout(function() {
-        // Delay the submit until the egg is finished hatching
+      // Delay the submit until the egg is finished hatching
+      setTimeout(() => {
+        const { correct, errors } = this.props,
+        { eggs } = this.state,
+        eggDrakeIndex = selectedEggIndex,
+        isChallengeComplete = correct + errors + 1 >= eggs.length;
         onSubmitEggForBasket(eggDrakeIndex, basketIndex, isSubmittedEggCorrect, isChallengeComplete);
       }, 3000);
```

Comment:
I think the delays after submitting are a little long - the animation of the egg moving and hatching finishes long before the egg submitted / egg accepted events fire.

Reply:
I'll experiment with shortening the delays.

Reply:
As I recall, the drakes used to fade automatically as soon as they reached the basket, and the delay was originally set to correspond to the end of the fade. Now that the drakes stay until the next click it makes sense to time the submit to the hatching of the drake, so I've reduced the delay accordingly.